packages/@blockprotocol/graph/src/graph-service.json

```diff
@@ -0,0 +1,690 @@
+{
+  "name": "graph",
+  "version": "0.2",
+  "coreVersion": "0.2",
+  "messages": [
+    {
+      "name": "createEntity",
+      "description": "Request to create an entity, expecting the created entity in response.",
+      "source": "block",
+      "respondedToByMessage": "createEntityResponse",
+      "sentOnInitialization": false,
+      "data": {
+        "type": "object",
+        "properties": {
+          "entityTypeId": {
+            "description": "The entityTypeId of the type of the entity to create",
+            "type": "string"
+          },
+          "properties": {
+            "description": "The properties of the entity to create",
+            "type": "object"
+          }
+        },
+        "required": ["entityTypeId", "properties"]
+      },
+      "errorCodes": []
+    },
+    {
+      "name": "createEntityResponse",
+      "description": "The response to a request to create an entity",
+      "source": "embedder",
+      "respondedToByMessage": null,
+      "sentOnInitialization": false,
+      "data": {
+        "type": "object",
+        "properties": {
+          "entity": {
+            "$ref": "https://blockprotocol.org/types/Entity"
+          }
+        },
+        "required": ["entity"]
+      },
+      "errorsCodes": ["FORBIDDEN", "INVALID_INPUT"]
+    },
+    {
+      "name": "updateEntity",
+      "description": "Request to update an entity, expecting the updated entity in response.",
+      "source": "block",
+      "respondedToByMessage": "updateEntityResponse",
+      "sentOnInitialization": false,
+      "data": {
+        "type": "object",
+        "properties": {
+          "entityId": {
+            "description": "The entityId of the entity to update",
+            "type": "string"
+          },
+          "properties": {
+            "description": "The properties of the entity to update, which will be merged with any other properties on the entity",
+            "type": "object"
+          }
+        },
+        "required": ["entityId", "properties"]
+      },
```

Comment:
My ideal scenario is that we have a single service definition - a longer version of this file - from which we can generate the types and the documentation (i.e. have markdown descriptions all over the place). 

I say that because it seems more accessible to have people write the thing in JSON, and generate other things from that, rather than having them write it in TypeScript.